s3-store/src/main/scala/geotrellis/store/s3/util/S3RangeReader.scala

```diff
@@ -35,26 +35,31 @@ import java.net.URI
  * @return A new instance of S3RangeReader.
  */
 class S3RangeReader(
-  request: GetObjectRequest,
+  request: => GetObjectRequest,
```

Comment:
It is by name parameter now, since otherwise, any `S3RangeReader` creation forced an immediate metadata queries.

Reply:
Also a reasonable question appears, should the client be a by name parameter and cached inside?

Reply:
That's a good question. Would that be beneficial to have? Because right now the `S3RangeReader` is more robust as it can work with more than one client.

Reply:
@jbouffard I am not sure that I follow you, the benefit would be that there would be no instance creation until the explicit RangeReader creation, in case `S3Client` was lazily created.